Index: dcmtk-3.6.0/ofstd/libsrc/CMakeLists.txt
===================================================================
--- dcmtk-3.6.0.orig/ofstd/libsrc/CMakeLists.txt
+++ dcmtk-3.6.0/ofstd/libsrc/CMakeLists.txt
@@ -1,5 +1,7 @@
 # create library from source files
 ADD_LIBRARY(ofstd ofcmdln ofconapp ofcond ofconfig ofconsol ofcrc32 ofdate ofdatime offname oflist ofstd ofstring ofthread oftime)
 
+target_link_libraries(ofstd pthread)
+
 # declare installation files
 INSTALL_TARGETS(${INSTALL_LIBDIR} ofstd)
Index: dcmtk-3.6.0/oflog/libsrc/CMakeLists.txt
===================================================================
--- dcmtk-3.6.0.orig/oflog/libsrc/CMakeLists.txt
+++ dcmtk-3.6.0/oflog/libsrc/CMakeLists.txt
@@ -8,5 +8,7 @@ ENDIF(WIN32 AND NOT CYGWIN)
 
 ADD_LIBRARY(oflog oflog apndimpl appender config consap factory fileap filter globinit hierarchy hierlock layout logger logimpl logevent loglevel loglog lloguser ndc ntelogap nullap objreg patlay pointer property rootlog sleep socketap sockbuff socket strhelp syncprims syslogap threads timehelp ${OFLOG_PLATFORM_LIBRARIES})
 
+target_link_libraries(oflog ofstd)
+
 # declare installation files
 INSTALL_TARGETS(${INSTALL_LIBDIR} oflog)
Index: dcmtk-3.6.0/dcmdata/libsrc/CMakeLists.txt
===================================================================
--- dcmtk-3.6.0.orig/dcmdata/libsrc/CMakeLists.txt
+++ dcmtk-3.6.0/dcmdata/libsrc/CMakeLists.txt
@@ -1,5 +1,7 @@
 # create library from source files
 ADD_LIBRARY(dcmdata cmdlnarg dcbytstr dcchrstr dccodec dcdatset dcddirif dcdicdir dcdicent dcdict dcdictzz dcdirrec dcelem dcerror dcfilefo dchashdi dcistrma dcistrmb dcistrmf dcistrmz dcitem dclist dcmetinf dcobject dcostrma dcostrmb dcostrmf dcostrmz dcpcache dcpixel dcpixseq dcpxitem dcrleccd dcrlecce dcrlecp dcrledrg dcrleerg dcrlerp dcsequen dcstack dcswap dctag dctagkey dctypes dcuid dcwcache dcvm dcvr dcvrae dcvras dcvrat dcvrcs dcvrda dcvrds dcvrdt dcvrfd dcvrfl dcvris dcvrlo dcvrlt dcvrobow dcvrof dcvrpn dcvrpobw dcvrsh dcvrsl dcvrss dcvrst dcvrtm dcvrui dcvrul dcvrulup dcvrus dcvrut dcxfer dcpath vrscan vrscanl)
 
+target_link_libraries(dcmdata ofstd oflog z)
+
 # declare installation files
 INSTALL_TARGETS(${INSTALL_LIBDIR} dcmdata)
Index: dcmtk-3.6.0/dcmdata/libi2d/CMakeLists.txt
===================================================================
--- dcmtk-3.6.0.orig/dcmdata/libi2d/CMakeLists.txt
+++ dcmtk-3.6.0/dcmdata/libi2d/CMakeLists.txt
@@ -1,5 +1,7 @@
 # create library from source files
 ADD_LIBRARY(libi2d i2d i2dplvlp i2djpgs i2dbmps i2dplsc i2dplnsc)
 
+target_link_libraries(libi2d ofstd oflog dcmdata)
+
 # declare installation files
 INSTALL_TARGETS(${INSTALL_LIBDIR} libi2d)
Index: dcmtk-3.6.0/dcmimgle/libsrc/CMakeLists.txt
===================================================================
--- dcmtk-3.6.0.orig/dcmimgle/libsrc/CMakeLists.txt
+++ dcmtk-3.6.0/dcmimgle/libsrc/CMakeLists.txt
@@ -1,5 +1,7 @@
 # create library from source files
 ADD_LIBRARY(dcmimgle dcmimage dibaslut diciefn dicielut didislut didispfn didocu digsdfn digsdlut diimage diinpx diluptab dimo1img dimo2img dimoimg dimoimg3 dimoimg4 dimoimg5 dimomod dimoopx dimopx diovdat diovlay diovlimg diovpln diutils)
 
+target_link_libraries(dcmimgle oflog ofstd dcmdata)
+
 # declare installation files
 INSTALL_TARGETS(${INSTALL_LIBDIR} dcmimgle)
Index: dcmtk-3.6.0/dcmjpeg/libsrc/CMakeLists.txt
===================================================================
--- dcmtk-3.6.0.orig/dcmjpeg/libsrc/CMakeLists.txt
+++ dcmtk-3.6.0/dcmjpeg/libsrc/CMakeLists.txt
@@ -4,5 +4,7 @@ INCLUDE_DIRECTORIES(${dcmjpeg_SOURCE_DIR
 # create library from source files
 ADD_LIBRARY(dcmjpeg ddpiimpl dipijpeg djcodecd djcodece djcparam djdecbas djdecext djdeclol djdecode djdecpro djdecsps djdecsv1 djdijg12 djdijg8 djdijg16 djeijg12 djeijg8 djeijg16 djencbas djencext djenclol djencode djencpro djencsps djencsv1 djrplol djrploss djutils)
 
+target_link_libraries(dcmjpeg ofstd oflog dcmimgle ijg8 ijg12 ijg16)
+
 # declare installation files
 INSTALL_TARGETS(${INSTALL_LIBDIR} dcmjpeg)
Index: dcmtk-3.6.0/dcmimage/libsrc/CMakeLists.txt
===================================================================
--- dcmtk-3.6.0.orig/dcmimage/libsrc/CMakeLists.txt
+++ dcmtk-3.6.0/dcmimage/libsrc/CMakeLists.txt
@@ -1,5 +1,7 @@
 # create library from source files
 ADD_LIBRARY(dcmimage diargimg dicmyimg dicoimg dicoopx dicopx dihsvimg dilogger dipalimg dipipng dipitiff diqtctab diqtfs diqthash diqthitl diqtpbox diquant diregist dirgbimg diybrimg diyf2img diyp2img)
 
+target_link_libraries(dcmimage ofstd oflog dcmdata tiff png dcmimgle)
+
 # declare installation files
 INSTALL_TARGETS(${INSTALL_LIBDIR} dcmimage)
Index: dcmtk-3.6.0/dcmnet/libsrc/CMakeLists.txt
===================================================================
--- dcmtk-3.6.0.orig/dcmnet/libsrc/CMakeLists.txt
+++ dcmtk-3.6.0/dcmnet/libsrc/CMakeLists.txt
@@ -1,5 +1,7 @@
 # create library from source files
 ADD_LIBRARY(dcmnet assoc cond dcasccff dcasccfg dccfenmp dccfpcmp dccfprmp dccfrsmp dccftsmp dccfuidh dcmlayer dcmtrans dcompat dimcancl dimcmd dimdump dimecho dimfind dimget dimmove dimse dimstore diutil dul dulconst dulextra dulfsm dulparse dulpres extneg lst dfindscu dcuserid scu scp)
 
+target_link_libraries(dcmnet ofstd oflog dcmdata)
+
 # declare installation files
 INSTALL_TARGETS(${INSTALL_LIBDIR} dcmnet)
Index: dcmtk-3.6.0/dcmwlm/libsrc/CMakeLists.txt
===================================================================
--- dcmtk-3.6.0.orig/dcmwlm/libsrc/CMakeLists.txt
+++ dcmtk-3.6.0/dcmwlm/libsrc/CMakeLists.txt
@@ -1,5 +1,7 @@
 # create library from source files
 ADD_LIBRARY(dcmwlm wlds wldsfs wlfsim wlmactmg)
 
+target_link_libraries(dcmwlm ofstd oflog dcmdata dcmnet)
+
 # declare installation files
 INSTALL_TARGETS(${INSTALL_LIBDIR} dcmwlm)
Index: dcmtk-3.6.0/dcmqrdb/libsrc/CMakeLists.txt
===================================================================
--- dcmtk-3.6.0.orig/dcmqrdb/libsrc/CMakeLists.txt
+++ dcmtk-3.6.0/dcmqrdb/libsrc/CMakeLists.txt
@@ -1,5 +1,7 @@
 # create library from source files
 ADD_LIBRARY(dcmqrdb dcmqrcbf dcmqrcbg dcmqrcbm dcmqrcbs dcmqrcnf dcmqrdbi dcmqrdbs dcmqropt dcmqrptb dcmqrsrv dcmqrtis)
 
+target_link_libraries(dcmqrdb ofstd oflog dcmdata dcmnet)
+
 # declare installation files
 INSTALL_TARGETS(${INSTALL_LIBDIR} dcmqrdb)
Index: dcmtk-3.6.0/dcmsr/libsrc/CMakeLists.txt
===================================================================
--- dcmtk-3.6.0.orig/dcmsr/libsrc/CMakeLists.txt
+++ dcmtk-3.6.0/dcmsr/libsrc/CMakeLists.txt
@@ -1,5 +1,7 @@
 # create library from source files
 ADD_LIBRARY(dcmsr dsrcitem dsrcodtn dsrcodvl dsrcomtn dsrcomvl dsrcontn dsrcsidl dsrdattn dsrdoc dsrdoctn dsrdoctr dsrdtitn dsrimgfr dsrimgtn dsrimgvl dsrnumtn dsrnumvl dsrpnmtn dsrreftn dsrscogr dsrsc3gr dsrscotn dsrsc3tn dsrscovl dsrsc3vl dsrsoprf dsrstrvl dsrtcodt dsrtcosp dsrtcotn dsrtcoto dsrtcovl dsrtextn dsrtimtn dsrtncsr dsrtree dsrtypes dsruidtn dsrwavch dsrwavtn dsrwavvl dsrxmlc dsrxmld dsriodcc dsrbascc dsrenhcc dsrcomcc dsrkeycc dsrmamcc dsrchecc dsrcolcc dsrprocc dsrxrdcc dsrspecc dsrmaccc dsrimpcc)
 
+target_link_libraries(dcmsr ofstd oflog dcmdata)
+
 # declare installation files
 INSTALL_TARGETS(${INSTALL_LIBDIR} dcmsr)
Index: dcmtk-3.6.0/dcmpstat/libsrc/CMakeLists.txt
===================================================================
--- dcmtk-3.6.0.orig/dcmpstat/libsrc/CMakeLists.txt
+++ dcmtk-3.6.0/dcmpstat/libsrc/CMakeLists.txt
@@ -1,5 +1,7 @@
 # create library from source files
 ADD_LIBRARY(dcmpstat dcmpstat dviface dvpsab dvpsabl dvpsal dvpsall dvpscf dvpscu dvpscul dvpsda dvpsdal dvpsfs dvpsga dvpsgal dvpsgl dvpsgll dvpsgr dvpsgrl dvpshlp dvpsib dvpsibl dvpsmsg dvpsov dvpsovl dvpspl dvpspl2 dvpspll dvpspr dvpsprt dvpsri dvpsril dvpsrs dvpsrsl dvpssp dvpsspl dvpssv dvpssvl dvpstat dvpstx dvpstxl dvpsvl dvpsvll dvpsvw dvpsvwl dvsighdl)
 
+target_link_libraries(dcmpstat ofstd oflog dcmdata dcmnet dcmimgle dcmqrdb dcmsr)
+
 # declare installation files
 INSTALL_TARGETS(${INSTALL_LIBDIR} dcmpstat)
Index: dcmtk-3.6.0/dcmjpls/libsrc/CMakeLists.txt
===================================================================
--- dcmtk-3.6.0.orig/dcmjpls/libsrc/CMakeLists.txt
+++ dcmtk-3.6.0/dcmjpls/libsrc/CMakeLists.txt
@@ -4,5 +4,7 @@ INCLUDE_DIRECTORIES(${dcmjpls_SOURCE_DIR
 # create library from source files
 ADD_LIBRARY(dcmjpls djcparam djdecode djencode djrparam djcodecd djutils djcodece)
 
+target_link_libraries(dcmjpls ofstd oflog dcmdata dcmimgle CharLS)
+
 # declare installation files
 INSTALL_TARGETS(${INSTALL_LIBDIR} dcmjpls)
